### PR TITLE
Feature: Create pages for vacations in the punch normal user side

### PR DESCRIPTION
**Pull request by HeitorMC**
Closes: #275 

### What?

For a "normal" user, create pages to see, request or cancel a vacation.

The user should be able to see all your own vacations requests
The normal user cannot cancel a not pending vacation

The form will have only the start date and end date.
The end date cannot be before the start date.
The start and the end date cannot be in the past.
The end date cannot be on the same day as the start date

### CODE REVIEW

File: config/locales/en/en.yml [hunk 1]
```diff
@@ -69,3 +69,6 @@ en:
   english_evaluation: "English evaluation"
   revenue_forecast: 'Revenue Forecast'
   revenue_forecast_warning: '* This feature was implemented on Sept/2022, and for this reason, data before this date are incomplete.'
+  vacations:
+    index:
+      are_you_sure_remove_request: 'Remove this vacation request?'
```
Comment (GabrielRMuller) [inline config/locales/en/en.yml:74]: [nit] need a new line

File: config/locales/miners/miner.en.yml [hunk 2]
```diff
@@ -66,3 +66,9 @@ en:
         open: Open
         closed: Closed
         merged: Merged
+    vacation:
+      status:
+        pending: Pending
+        approved: Approved
+        refused: Refused
+        canceled: Canceled
```
Comment (GabrielRMuller) [inline config/locales/miners/miner.en.yml:74]: in this translation file we're calling it `canceled` with a single L, everywhere else we're calling `cancelled` with two L's, maybe change it to `cancelled` here as well?

File: spec/controllers/vacations_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  before do
+    allow(controller).to receive(:authenticate_user!)
+    allow(controller).to receive(:current_user).and_return(user)
+  end
+
+  describe 'GET index' do
+    let!(:vacation) { create(:vacation, user: user) }
+
+    it "returns all user's vacations" do
```
Comment (GabrielRMuller) [inline spec/controllers/vacations_controller_spec.rb:14]: this spec is only checking if the request is successful, but the `it` suggests it's testing if all the user's vacations is being returned. I'd suggest renaming this one to reflect what it's testing and then add a new one to check if we're getting the vacations.

File: spec/controllers/vacations_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      get :index
+
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe 'GET show' do
+    let(:vacation) { create(:vacation, user: user) }
+
+    it "returns a specific user vacation" do
```
Comment (GabrielRMuller) [inline spec/controllers/vacations_controller_spec.rb:24]: similar to my comment on index

File: app/models/vacation.rb [hunk 5]
```diff
@@ -10,9 +11,33 @@ class Vacation < ApplicationRecord
...
+    @starting_day = date.presence
+    mount_start_date
+    date
+  end
+
+  def ending_day=(date)
+    @ending_day = date.presence
+    mount_end_date
+    date
+  end
```
Comment (GabrielRMuller) [inline app/models/vacation.rb:32]: sorry I didn't quite understood what are these methods doing and where they're being used, could you elaborate a little bit on them?

File: app/controllers/vacations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      vacation_to_cancel.update!(status: :cancelled)
+      redirect_to vacations_path, notice: I18n.t(:notice, scope: "flash.vacation.cancel")
+    else
+      redirect_to vacations_path, alert: I18n.t(:alert, scope: "flash.vacation.cancel")
+    end
+  end
+
+  private
+
+  def scopped_vacations
```
[the remark this answers is not in the document]
Comment (FabioSaito) [reply app/controllers/vacations_controller.rb]: Fixed! Thanks

File: app/views/vacations/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<%= link_to 'Solicitar Férias', new_vacation_path %>
```
[the remark this answers is not in the document]
Comment (FabioSaito) [reply app/views/vacations/index.html.erb]: good catch, i misssed this one

File: spec/controllers/vacations_controller_spec.rb (hunk 3 above)
Comment (FabioSaito) [reply spec/controllers/vacations_controller_spec.rb]: Could you review again? I just updated these tests

File: app/models/vacation.rb (hunk 5 above)
Comment (FabioSaito) [reply app/models/vacation.rb]: I reimplemented in a better way now, it was a mistake

**Review by GabrielRMuller — APPROVED** [review]
nice specs!

**Pull request merged**